backend/src/resolvers.ts

```diff
@@ -0,0 +1,85 @@
+import PostDatasource from "./datasources/post.datasource";
+import UserDatasource from "./datasources/user.datasource";
+import Post from "./entities/post.entity";
+import User from "./entities/user.entity";
+
+const resolvers = {
+    Query: {
+        posts: async (_source, _args, context) => {
+            const postDatasource: PostDatasource = context.dataSources.postDatasource;
+
+            return (await postDatasource.getPosts()).map(post => {
+                return {
+                    ...post, votes:post.voters.length   
```

Comment:
`votes` should be a type resolver:
```js
Post: {
  votes: (post) => post.voters.length
}
```
That way you avaid this computation when the client does not request `votes`.